### PR TITLE
fsync() block devices before detaching them during shutdown

### PR DESCRIPTION
**Pull request by poettering**
In a way a follow-up for cae1e8fb88c5a6b0960a2d0be3df8755f0c78462

### CODE REVIEW

File: src/shutdown/umount.c [hunk 1]
```diff
@@ -387,8 +387,15 @@ static int delete_loopback(const char *device) {
...
-        if (fd < 0)
+        if (fd < 0) {
+                log_debug_errno(errno, "Failed to open loopback device %s: %m", device);
                 return errno == ENOENT ? 0 : -errno;
+        }
+
+        /* The loopback device deoesn't sync in-flight blocks when we clear the fd, hence sync explicitly
+         * first */
+        if (fsync(fd) < 0)
+                log_debug_errno(errno, "Failed to sync loop block device, ignoring %s: %m", device);
```
Comment (marcosfrm) [inline src/shutdown/umount.c]: `log_debug_errno(errno, "Failed to sync loop block device %s, ignoring: %m", device);` ?

Comment (poettering) [reply src/shutdown/umount.c]: true!

**Comment by poettering** [comment]
force pushed new version, only changes is the one suggested by @marcosfrm

File: src/shutdown/umount.c [hunk 2]
```diff
@@ -387,8 +387,15 @@ static int delete_loopback(const char *device) {
         assert(device);
 
         fd = open(device, O_RDONLY|O_CLOEXEC);
-        if (fd < 0)
+        if (fd < 0) {
+                log_debug_errno(errno, "Failed to open loopback device %s: %m", device);
                 return errno == ENOENT ? 0 : -errno;
+        }
+
+        /* The loopback device deoesn't sync in-flight blocks when we clear the fd, hence sync explicitly
```
Comment (keszybz) [inline src/shutdown/umount.c]: typo

**Review by keszybz — COMMENTED** [review]
LGTM.

File: src/shutdown/umount.c (hunk 2 above)
Comment (poettering) [reply src/shutdown/umount.c]: fixed

**Pull request merged**